src/script/script.cpp

```diff
@@ -305,28 +327,50 @@ std::string CScript::ToString() const
 
 CScript::ScriptType CScript::GetType() const
 {
-    if (this->IsPayToPublicKeyHash())
-        return CScript::P2PKH;
-    if (this->IsPayToScriptHash())
-        return CScript::P2SH;
+    if (IsPayToPublicKeyHash())
+        return CScript::ScriptType::P2PKH;
+    if (IsPayToScriptHash())
+        return CScript::ScriptType::P2SH;
+    if (IsPayToPublicKey())
+        return CScript::ScriptType::P2PK;
     // We don't know this script type
-    return CScript::UNKNOWN;
+    return CScript::ScriptType::UNKNOWN;
 }
 
 uint160 CScript::AddressHash() const
 {
     // where the address bytes begin depends on the script type
-    int start;
-    if (this->IsPayToPublicKeyHash())
+    int start, size;
+    bool isPayToPublicKey = false;
+    if (IsPayToPublicKeyHash())
+    {
         start = 3;
-    else if (this->IsPayToScriptHash())
+        size = 20;
+    }
+    else if (IsPayToScriptHash())
+    {
         start = 2;
+        size = 20;
+    }
+    else if (IsPayToPublicKey())
+    {
+        start = 1;
+        size = 33;
+        isPayToPublicKey = true;
+    }
     else {
         // unknown script type; return zeros (this can happen)
         vector<unsigned char> hashBytes;
         hashBytes.resize(20, 0);
         return uint160(hashBytes);
```

Comment:
How about
```suggestion
        return uint160(std::vector<unsigned char>(size));
```
where `size` is initialized to `20` when it's declared?

Reply:
Make sense, more compact. Done